### PR TITLE
Sparkline, Bullet: Use contentTemplate in tooltip by default customizeTooltip instead

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

File: testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js [hunk 1]
```diff
@@ -47,77 +80,91 @@ const environment = {
...
+            }
+        }, options)).dxBullet('instance');
     }
 };
 
-QUnit.module('Tooltip creating', environment);
+QUnit.module('Bullet tooltip', environment);
 
-QUnit.test('Enabled tooltip', function(assert) {
+QUnit.test('Tooltip constructor should accept valid parems when tooltip enabled', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js]: plz, check everywhere
```suggestion
QUnit.test('Tooltip constructor should accept valid params when tooltip is enabled', function(assert) {
```

File: testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js [hunk 2]
```diff
@@ -47,77 +80,91 @@ const environment = {
...
-    assert.equal(bullet._tooltip.update.callCount, 1, 'update is called');
-    assert.ok($.isFunction(bullet._tooltip.update.lastCall.args[0].customizeTooltip));
-    bullet._tooltip.update.lastCall.args[0].customizeTooltip = {};
-    assert.deepEqual(bullet._tooltip.update.lastCall.args[0], {
-        enabled: false,
-        font: {
-            size: 12
-        },
-        customizeTooltip: {}
+QUnit.test('Update method shoul accept valid params when tooltip enabled and no data', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js]: plz, check everywhere
```suggestion
QUnit.test('Update method should accept valid params when tooltip is enabled and no data', function(assert) {
```

File: testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js [hunk 3]
```diff
@@ -157,43 +203,71 @@ QUnit.test('dxBullet get TooltipFormatObject', function(assert) {
     }, {}]);
 });
 
-QUnit.test('Default Tooltip text', function(assert) {
+QUnit.test('Default template should be used when customizeTooltip is nota defined', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.sparklines/bulletTooltip.tests.js]: plz,  check everywhere
```suggestion
QUnit.test('Default template should be used when customizeTooltip is not specified', function(assert) {
```

File: js/viz/sparklines/base_sparkline.js [hunk 4]
```diff
@@ -35,35 +38,35 @@ function pointerHandler({ data }) {
     that._showTooltip();
 }
 
-function generateDefaultCustomizeTooltipCallback(fontOptions, rtlEnabled) {
-    const lineSpacing = fontOptions.lineSpacing;
-    const lineHeight = ((lineSpacing !== undefined && lineSpacing !== null) ? lineSpacing : DEFAULT_LINE_SPACING) + fontOptions.size;
+function getDefaultTemplate({ lineSpacing, size }, textAlign) {
+    const lineHeight = `${((lineSpacing !== undefined && lineSpacing !== null) ? lineSpacing : DEFAULT_LINE_SPACING) + size}px`;
```
Comment (ksercs) [inline js/viz/sparklines/base_sparkline.js]:
```suggestion
    const lineHeight = `${((lineSpacing ?? DEFAULT_LINE_SPACING) + size}px`;
```

File: js/viz/sparklines/base_sparkline.js [hunk 5]
```diff
@@ -272,12 +275,35 @@ BaseSparkline.prototype._setTooltipRendererOptions = function() {
...
+
+    if(isFunction(options.customizeTooltip)) {
+        this._tooltip.update(options);
+
+        const formatObject = this._getTooltipData();
+        const customizeResult = options.customizeTooltip.call(formatObject, formatObject) ?? {};
+
+        if(!('html' in customizeResult) && !('text' in customizeResult)) {
+            defaultTemplateNeeded = true;
+        }
```
Comment (ksercs) [inline js/viz/sparklines/base_sparkline.js]: we can get rid of the condition
```javascript
defaultTemplateNeeded = !('html' in customizeResult) && !('text' in customizeResult);
```

File: js/viz/sparklines/base_sparkline.js [hunk 6]
```diff
@@ -272,12 +275,33 @@ BaseSparkline.prototype._setTooltipRendererOptions = function() {
...
+        const contentTemplateOptions = defaultContentTemplate ? { contentTemplate: defaultContentTemplate } : {};
+        const optionsToUpdate = _extend(contentTemplateOptions, options, {
+            enabled: options.enabled && this._isTooltipEnabled()
+        });
+        this._tooltip.update(optionsToUpdate);
+    }
+};
+
+BaseSparkline.prototype._getDefaultTooltipTemplate = function(options) {
+    let defaultTemplateNeeded = false;
```
Comment (Krijovnick) [inline js/viz/sparklines/base_sparkline.js]: The `defaultTemplateNeeded` can be declared with true, and than block with `else` can be removed

**Review by Krijovnick — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**